aws_topology/stackstate_checks/aws_topology/flowlogs.py

```diff
@@ -0,0 +1,168 @@
+        for i in range(0, len(files), self.MAX_S3_DELETES):
+            try:
+                self.log.info(
+                    "Deleting {} files from S3 bucket {}".format(len(files[i : i + self.MAX_S3_DELETES]), bucket_name)
+                )
+                client.delete_objects(
+                    Bucket=bucket_name,
+                    Delete={"Objects": files[i : i + self.MAX_S3_DELETES], "Quiet": True},
+                )
+            except Exception as e:
+                self.log.exception(e)
+                self.agent.warning("CloudtrailCollector: Deleting s3 files failed")
+
+    def process_files(self, client, bucket_name, files, nwinterfaces):
+        traffic = {}
+        for file in files:
+            s3_body = client.get_object(Bucket=bucket_name, Key=file).get('Body')
+            self._delete_files(client, bucket_name, [{"Key": file}])
+            with self._get_stream(s3_body) as data:
+                first = True
+                for line in data:
+                    line = line.decode('ascii').strip()
+                    if first:
+                        first = False
+                        flds = line.split(' ')
+                    else:
+                        vals = line.split(' ')
+                        log = {fld: val for (fld, val) in zip(flds, vals)}
+                        if log.get('log-status', 'NODATA') != 'NODATA':
+                            nwitf = nwinterfaces.get(log["interface-id"], None)
+                            pip = nwitf["PrivateIpAddress"]
+                            dir = "unknown"
+                            ip = traffic.get(pip)
+                            if ip is None:
+                                ip = traffic[pip] = {"in": {}, "out": {}, "vpc": nwitf["VpcId"]}
+                            private = None
+                            other_ip = ""
+                            if pip == log["srcaddr"]:
+                                other_ip = log["dstaddr"]
+                                private = is_private(other_ip)
+                                dir = "out"
+                            elif pip == log["dstaddr"]:
+                                other_ip = log["srcaddr"]
+                                private = is_private(other_ip)
+                                dir = "in"
+                            if private:
+                                rec = ip.get(dir, {}).get(other_ip)
+                                if not rec:
+                                    res = {
+                                        "packets": int(log["packets"]),
+                                        "bytes": int(log["bytes"])
+                                    }
+                                    ip[dir][other_ip] = res
+                                else:
+                                    res["bytes"] += int(log["bytes"])
+                                    res["packets"] += int(log["packets"])
+
+        # process traffic
+        connections = set()
```

Comment:
If you use the model above with the connection list, this bottom part will also be cleaned up quite a bit. I think you can get rid of `add_to_connections` and the `for nwi in traffic` loop.

Reply:
I have kept nwitf around for future improvement that use their publicip / privateip relation